presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -398,6 +416,10 @@ else if (!result.getBlocked().isDone()) {
             }
 
             scheduling.set(false);
+            if (isPartialResultsEnabled(session)) {
```

Comment:
This check might not be necessary given we already check the same before adding the tasks to tracker.